pkg/cluster/sync_test.go

```diff
@@ -177,6 +216,41 @@ func TestConnectionPoolerSynchronization(t *testing.T) {
 			defaultInstances: 1,
 			check:            objectsAreDeleted,
 		},
+		{
+			subTest: "delete only master if not needed",
```

Comment:
Judging form the specification, a master connection pooler here was not created, which somehow contradicts with the test description. What is the point of this test?